### DISCUSSION TITLE
Backend RAG support,  bulk documents.

### DISCUSSION THREAD

**Discussion by aamir-gmail** [open-webui/open-webui#1803] [General]
What  would make a great addition to this , would be a RAG backed to load  bulk documents . The way to do this would simply  mount an external volume  to the docker image. This way initial emedding are created when launching the docker for the first time .Then have a file watcher  to  load up any new document which are added to the external directory that is without doing this manually all this time or restarting the docker image. . All documents are avaiable to all users of Web-UI for RAG use. While the other option of loading documents through the Web-UI is still there however private to that users only.

When using this feature UI should provide the sources as links as to which particular document it is getting the information from.

Thank you.

**Comment by aamir-gmail** [comment]
Adding further comments , might I suggest moving  "chroma.sqlite3" to full enterprise class vector DB like "Milvus" or "Weaviate". The bulk loading 50 GB documernts (text , Docs or PDF ) might prove too much for embeded DB like "chroma.sqlite3".

**Comment by joemarlo** [comment]
I second this. Any features that makes bulk uploading of docs or adding external files easier would be great. I assume this is partly possible by programmatically putting documents into `/data/docs` but I haven't tested it. I would love to be able to point it at all my private GitHub repos and be able to ask general questions. Or point it at my Google Docs. Or point it at Slack, etc...

**Reply by aamir-gmail** [reply]
I have tried 'data/docs' I am using the docker version where 'data/docs' is mounted to external disk '/disk1/..." . At least for me it does not work. Even if it did I do not think that embedded "chroma.sqlite3" would be upto the challange. In the docker image using supervisior 
http://supervisord.org/  one would launch two procsses , one for vector DB like "Milvus" or "Weaviate" and the other for Open-web-ui.

Supervisor is quiets capable of handling two or more procesees and restart as required. Open-web-Ui is a great project well throught of and designed. Few minor improvements would make a really good product.

**Comment by thohemp** [comment]
@aamir-gmail Any updates on this?
